Dangerfile

```diff
@@ -22,3 +22,11 @@ Please run `bundle install` to make sure they match.
 }
   fail(message)
 end
+
+# Lint with Rubocop and report violations inline in GitHub
+github.dismiss_out_of_range_messages # This way, fixed violations should go
+rubocop.lint(
+  files: git.modified_files + git.added_files,
+  inline_comment: true,
+  fail_on_inline_comment: true # Make the inline comments failures
```

Comment:
I decided to configure it strict, so that we won't miss lint violations because they'll make the build fail.

Reply:
This list of files miss some cases:

* It forgets to remove deleted files. One would think that just not listing the deleted files in the list will just not include it, but I think I've seen in the past some cases where the deleted files were also present in the modified files list, at least I have example of Dangerfiles I used in past projects where I explicitly removed them from the list to be sure, so better safe than sorry.
```
  files: (git.modified_files - git.deleted_files) + git.added_files
```

* It also doesn't handle renamed files. Those are a more complicated beast, because `renamed_files` is not just a flat array, but a list of old name => new name pairs, so we need to `map` to get the proper list. (I sadly didn't manage to dig example of past Dangerfiles on how I did that in past project, so probably needs some digging to figure out the right way)

Reply:
Thanks @AliSoftware. I'll test these out in #247...

Reply:
Reopened #247 to push a file renaming and test this out, and indeed it appears that it doesn't lint the file if it has been renamed as I feared. See https://github.com/wordpress-mobile/release-toolkit/pull/247#issuecomment-831972829 for more details.

Reply:
Thanks a lot for looking into this @AliSoftware 🙇 .

I started working on this because I hoped it would have been a quick fix, but it looks like it isn't 😞 .

I don't have bandwidth to look into it right now, but I definitely want to see this happen here. Feel free to take over the PR if you have time / ideas on how to fix the setup. Otherwise, I'll get to this likely past mid-May.

Reply:
Submitted a patch to `danger-rubocop` FYI: https://github.com/ashfurrow/danger-rubocop/pull/50

Reply:
(…which has already been merged and released into a new version of danger-rubocop! 🎉 )